Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromCondDB.cc

```diff
@@ -37,146 +36,182 @@ GEMGeometryBuilderFromCondDB::build(const std::shared_ptr<GEMGeometry>& theGeome
-
-    tranStart = rgeo.tranStart( id );
-    shapeStart = rgeo.shapeStart( id );
-    rotStart = rgeo.rotStart( id );
-    strStart = rgeo.strStart( id );
-    name = *( strStart );
-
-    Surface::PositionType pos(*(tranStart)/cm, *(tranStart+1)/cm, *(tranStart+2)/cm );
-    // CLHEP way
-    Surface::RotationType rot(*(rotStart+0), *(rotStart+1), *(rotStart+2),
-                              *(rotStart+3), *(rotStart+4), *(rotStart+5),
-                              *(rotStart+6), *(rotStart+7), *(rotStart+8));
+  std::map<GEMDetId, GEMSuperChamber*> superChambers;
+  std::map<GEMDetId, GEMChamber*> chambers;
+  std::map<GEMDetId, GEMEtaPartition*> partitions;
     
-    float be = *(shapeStart+0)/cm;
-    float te = *(shapeStart+1)/cm;
-    float ap = *(shapeStart+2)/cm;
-    float ti = *(shapeStart+3)/cm;
-    float nstrip = *(shapeStart+4);
-    float npad = *(shapeStart+5);
-    //  TrapezoidalPlaneBounds* 
-    Bounds* bounds = new TrapezoidalPlaneBounds( be, te, ap, ti );
-
-    std::vector<float> pars;
-    pars.emplace_back(be); //b/2;
-    pars.emplace_back(te); //B/2;
-    pars.emplace_back(ap); //h/2;
-    pars.emplace_back(nstrip);
-    pars.emplace_back(npad);
-    
-    GEMEtaPartitionSpecs* epSpecs = new GEMEtaPartitionSpecs( GeomDetEnumerators::GEM, name, pars );
-      
-    //Change of axes for the forward
-    Basic3DVector<float> newX( 1., 0., 0. );
-    Basic3DVector<float> newY( 0., 0., 1. );
-    //      if (tran[2] > 0. )
-    newY *= -1;
-    Basic3DVector<float> newZ( 0., 1., 0. );
-    rot.rotateAxes( newX, newY, newZ );	
-  
-    BoundPlane* bp = new BoundPlane( pos, rot, bounds );
-    ReferenceCountingPointer<BoundPlane> surf( bp );
-    GEMEtaPartition* gep = new GEMEtaPartition( gemid, surf, epSpecs );
-    LogDebug("GEMGeometryBuilder") << "GEM Eta Partition created with id = " << gemid
-				   << " and added to the GEMGeometry" << std::endl;
-    theGeometry->add(gep);
+  for( unsigned int id = 0; id < detids.size(); ++id ){  
+    GEMDetId gemid( detids[id] );
+    LogDebug("GEMGeometryBuilderFromDDD") <<"GEMGeometryBuilderFromDDD adding " << gemid << std::endl;
     
-    std::list<GEMEtaPartition *> gepList;
-    if( m_chids.find( chid ) != m_chids.end()) {
-      gepList = m_chids[chid];
-    }
-    gepList.emplace_back(gep);
-    m_chids[chid] = gepList;
-  }
-  
-  // Create the GEMChambers and store them on the Geometry 
-
-  for( const auto& ich : m_chids ) {
-    GEMDetId chid = ich.first;
-    std::list<GEMEtaPartition * > gepList = ich.second;
-
-    // compute the overall boundplane. At the moment we use just the last
-    // surface
-    BoundPlane* bp = nullptr;
-    for( const auto& gep : gepList ) {
-      const BoundPlane& bps = ( *gep ).surface();
-      bp = const_cast<BoundPlane *>( &bps );
+    if (gemid.roll() == 0){
+      if (gemid.layer() == 0){
+	GEMSuperChamber* gsc = buildSuperChamber( rgeo, id, gemid );
+	superChambers.insert(std::pair<GEMDetId, GEMSuperChamber*>(gemid, gsc));
```

Comment:
use `emplace(gemid,gsc)` (and also below)

Reply:
yep